pkg/server/status.go

```diff
@@ -1461,18 +1461,73 @@ func (s *statusServer) Stacks(
 	return stacksLocal(req)
 }
 
+func (s *statusServer) processGoroutineProfilesFromAllNodes(
+	ctx context.Context, response profDataResponse,
+) ([]byte, error) {
+	res := bytes.NewBuffer(nil)
+	for nodeID, pd := range response.profDataByNodeID {
+		if len(pd.data) == 0 && pd.err == nil {
+			log.Warningf(ctx, "no stacks collected for node %d", nodeID)
```

Comment:
Is this useful? As is, the caller won't know the information is incomplete, so the UX for making this discoverable is poor. Consider adding the same plumbing for errors back into the response here. I see that the caller of this method gets to return a `JSONResponse`, so it would be straightforward to return an additional slice `[]error` from this method and put it into the `JSONResponse{Errors: errs.Map(func(err error) string { return err.Error() })}` (this doesn't exist but you get the idea).

Reply:
This makes sense but I'm not sure how we'd surface these `Errors` when writing the data to the http.ResponseWriter. In the case of all profiles except goroutine profiles with labels we redirect the user to a flamegraph with the serialized profile data in the response body. I was thinking of using the `Warning` response header - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Warning but this is marked as deprecated.

We could do the following:
- Set the `Comments` field in the `Profile` to contain the partial errors so that when the user switches to the `Peek` view they can notice the missing nodes. This would cover cpu profiles and goroutine profiles without labels.
- When goroutine profiles are set with labels we can log the partial errors at the top of the file that is queued to download.

Let me know if this is not what you had in mind!